src/prognostic_equations/implicit/implicit_solver.jl

```diff
@@ -804,8 +804,11 @@ function update_implicit_equation_jacobian!(A, Y, p, dtγ)
                 ᶜdiffusion_h_matrix_scaled,
                 ᶜdiffusion_h_matrix,
             )
-            @. ∂ᶜρq_err_∂ᶜρ =
-                dtγ * ᶜtridiagonal_matrix_scalar ⋅ DiagonalMatrixRow(-(ᶜq) / ᶜρ)
+            if q_name ∉ (@name(q_liq), @name(q_ice), @name(q_rai), @name(q_sno))
+                @. ∂ᶜρq_err_∂ᶜρ =
```

Comment:
You might need to set it to zero for the `else` case, as this is initialized with `similar` I think. You can check if you get some error now (which is likely not reproducible).

Reply:
Fair, I'll double check this but REPL tests suggested that this variable was indeed zero prior to the assignment based on `tracer_info`.

Reply:
I'm just worried it will be assigned a random number, and in most case it will actually be zero, so we don't see the problem.